azure_arc_data_jumpstart/aks/ARM/artifacts/DeploySQLMIADAuth.ps1

```diff
@@ -0,0 +1,276 @@
+$podname = "${sqlMIName}-0"
+kubectl exec $podname -c arc-sqlmi -n arc -- /opt/mssql-tools/bin/sqlcmd -S localhost -U $env:AZDATA_USERNAME -P "$env:AZDATA_PASSWORD" -Q "CREATE LOGIN [${domain_netbios_name}\$env:adminUsername] FROM WINDOWS"
+Write-Host "Created Windows user account ${domain_netbios_name}\$env:AZDATA_USERNAME in SQLMI instance."
+
+kubectl exec $podname -c arc-sqlmi -n arc -- /opt/mssql-tools/bin/sqlcmd -S localhost -U $env:AZDATA_USERNAME -P "$env:AZDATA_PASSWORD" -Q "EXEC master..sp_addsrvrolemember @loginame = N'${domain_netbios_name}\$env:adminUsername', @rolename = N'sysadmin'"
+Write-Host "Granted sysadmin role to user account ${domain_netbios_name}\$env:AZDATA_USERNAME in SQLMI instance."
+
+# Downloading demo database and restoring onto SQL MI
+Write-Host "`n"
+Write-Host "Downloading AdventureWorks database for MS SQL... (1/2)"
+kubectl exec $podname -n arc -c arc-sqlmi -- wget https://github.com/Microsoft/sql-server-samples/releases/download/adventureworks/AdventureWorks2019.bak -O /var/opt/mssql/data/AdventureWorks2019.bak 2>&1 | Out-Null
+Write-Host "Restoring AdventureWorks database for MS SQL. (2/2)"
+kubectl exec $podname -n arc -c arc-sqlmi -- /opt/mssql-tools/bin/sqlcmd -S localhost -U $Env:AZDATA_USERNAME -P "$Env:AZDATA_PASSWORD" -Q "RESTORE DATABASE AdventureWorks2019 FROM  DISK = N'/var/opt/mssql/data/AdventureWorks2019.bak' WITH MOVE 'AdventureWorks2017' TO '/var/opt/mssql/data/AdventureWorks2019.mdf', MOVE 'AdventureWorks2017_Log' TO '/var/opt/mssql/data/AdventureWorks2019_Log.ldf'" 2>&1 $null
+Write-Host "Restoring AdventureWorks database completed."
+
+# Retrieving SQL MI connection endpoint
+$sqlmiEndPoint = kubectl get SqlManagedInstance $sqlMIName -n arc -o=jsonpath='{.status.endpoints.primary}'
+
+Write-Host "SQL Managed Instance with AD authentication endpoint: $sqlmiEndPoint"
+
+# Get public ip of the SQLMI endpoint
+$nodeRG = (az aks show --name $Env:clusterName -g $Env:resourceGroup --query "nodeResourceGroup")
+$lbName = "kubernetes"
+$lbrule = (az network lb rule list -g $nodeRG --lb-name  $lbName --query "[?contains(id, '$sqlmi_port')]") | ConvertFrom-Json
+if ($null -eq $lbrule -or $lbrule.Count -le 0)
+{
+    Write-Host "Could not find LoadBalancer for SQLMI."
+    Stop-Transcript
+    Exit
+}
+
+$frontendIpConfId = $lbrule.frontendIpConfiguration.id
+$pubipid = (az network lb frontend-ip list --lb-name $lbName -g $nodeRG --query "[?id=='$frontendIpConfId'].{id:publicIpAddress.id}") | ConvertFrom-Json
+$publicIp =  (az network public-ip show --ids $pubipid.id --query "ipAddress").trim('"')
+Write-Host "SQLMI public ip address $publicIp"
+
+# Create DNS record
+Add-DnsServerResourceRecord -ComputerName $dcInfo.HostName -ZoneName $dcInfo.Domain -A -Name $sqlMIName -AllowUpdateAny -IPv4Address $publicIp -TimeToLive 01:00:00 -AgeRecord
+Write-Host "Creted SQLMI DNS A record with public ip address $publicIp"
+
+# Write endpoint information in the file
+$filename = "SQLMI-ADAuth-Endpoints.txt"
```

Comment:
can we use the same endpoints file instead of generating a new one ?

Reply:
Yes

Reply:
Fixed